doc/papers/joss/paper.md

```diff
@@ -0,0 +1,139 @@
+on a compute node or process, where $G$ is a local element restriction operation that
+localizes DOFs based on the elements, $B$ defines the action of the basis functions
+(or their gradients) on the nodes, and $D$ is the user-defined pointwise function
+describing the physics of the problem at the quadrature points, also called the
+QFunction. QFunctions, which can either be defined by the user or selected from a
+gallery of available built-in functions in the library, are pointwise functions
+that do not depend on element resolution, topology, or basis degree (selectable
+at run time). This easily allows $hp$-refinement studies (where $h$ commonly denotes the average element size and $p$ the polynomial degree of the basis functions in 1D) and $p$-multigrid solvers. libCEED also supports composition of different operators for multiphysics problems and mixed-element meshes (see Fig. \ref{fig:schematic}).
+
+![A schematic of element restriction and basis applicator operators for
+elements with different topology. This sketch shows the independence of QFunctions
+(in this case representing a Laplacian) element resolution, topology, or basis degree.\label{fig:schematic}](img/QFunctionSketch.pdf)
+
+LibCEED is a C99 library with Fortran77 and Python interfaces. The Python interface was developed using the C Foreign Function Interface (CFFI) for Python. CFFI allows to reuse most of the C declarations and requires only a minimal adaptation of some of them. The C and Python APIs are mapped in a nearly 1:1 correspondence. For instance, a ``CeedVector`` object is exposed as ``libceed.Vector`` in Python, and may reference memory that is also accessed via Python arrays from the NumPy [@NumPy] or Numba [@Numba] packages, for handling host or device memory (when interested in GPU computations with CUDA). Flexible pointer handling in libCEED makes it easy to provide zero-copy host and (GPU) device support for any desired Python array container. The interested reader can find more details on libCEED's Python interface in [@libceed-paper-proc-scipy-2020].
+
+To achieve high performance, libCEED can take advantage of a tensor-product
+finite-element basis and quadrature rule to apply the action of the basis
+operator $B$ or, alternatively, efficiently operate on bases that are defined
+on arbitrary-topology elements. Furthermore, the algebraic decomposition described in
+Eq. (\ref{eq:decomposition}) can represent either linear/nonlinear or
+symmetric/asymmetric operators and exposes opportunities for device-specific
+optimizations.
+
+![libCEED is a low-level API for finite element codes, that has specialized implementations
+(backends) for heterogeneous architectures.\label{fig:libCEEDBackends}](img/libCEEDBackends.pdf)
+
+Fig. \ref{fig:libCEEDBackends} shows a subset of the backend implementations (backends) available in libCEED and its role, as a low-level library that allows a wide variety of applications to share highly optimized discretization kernels.
```

Comment:
Will the figure need an update?

Reply:
Updated with your latest version. Thanks.